### DISCUSSION TITLE
Week 1 Kelas Xion : Adakah padanan library(dplyr) di phyton?

### DISCUSSION THREAD

**Discussion by ammaralwandi97** [teamalgoritma/community#807] [Q&A] [answered]
Bahasa sehari-hari saya R, skrg sedang mendalami Phyton. Untuk Data Wragling, adakah library dari Phyton yang setara dengan dplyr?

**Comment by IrfanChairurrachman** [comment, marked as answer]
Dear @ammaralwandi97 ,

`dplyr` adalah library untuk membantu pengolahan dataframe di R. Di Python library yang sangat popular untuk mengolah dataframe adalah `pandas`. Fungsi-fungsi di `dplyr` secara umum telah tercover juga di `pandas`. 

Namun, konsep di `dplyr` yang sangat powerfull dan jarang dilihat di penggunaan `pandas` adalah konsep `piping %>%` untuk menyambungkan tahapan-tahapan dalam 1 proses. Di `pandas` terdapat konsep `piping` menggunakan fungsi `.pipe()`. Contoh implementasi `piping` di pandas dapat dilihat pada link berikut:

- [Dokumentasi piping](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.pipe.html)
- [Contoh implementasi](https://www.kdnuggets.com/2021/01/cleaner-data-analysis-pandas-pipes.html)

Sekian jawaban dari kami. Bila ternyata informasi ini dapat menjawab pertanyaan Anda, Anda dapat “Mark as answer” jawaban ini.

**Reply by ammaralwandi97** [reply]
lalu apa kelebihan phyton dibandingkan R jika fungsinya sama saja? mohon penjelasan yang komprehensif
